src/lib/middleware/maintenance-middleware.ts

```diff
@@ -0,0 +1,24 @@
+import { IUnleashConfig } from '../types';
+import { Request } from 'express';
+
+const maintenanceMiddleware = ({
+    getLogger,
+    flagResolver,
+}: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>): any => {
+    const logger = getLogger('/middleware/maintenance-middleware.ts');
+    logger.debug('Enabling Maintenance middleware');
+
+    return async (req: Request, res, next) => {
+        try {
+            const writeMethod = ['POST', 'PUT', 'DELETE'].includes(req.method);
+            if (writeMethod && flagResolver.isEnabled('maintenance')) {
+                res.status(503).send({});
+            }
+        } catch (error) {
+            logger.error(error);
+        }
+        next();
```

Comment:
we may get an error here with double response sending. One from res.status(503) and one from whatever is next. it's safer to either res.status in middleware or next but not both.

Reply:
In other words we don't return after line 15 with res.status and the function continues.

Reply:
I believe send() ends the function. If I would just set res.status() then it would continue.

Reply:
without return it continues

Reply:
Very often it leads to this error message in the console: "Can't set headers after they are sent to the client"

Reply:
try this example reproducing the problem
```js
const express = require("express");
const app = express();

app.use((req, res, next) => {
    res.send("middleware");
    next();
});
app.get('/', (req, res, next) => {
    res.send("handler");
});

app.listen(3000);
```

Reply:
Yep, seems to be so. Should be fine now.